### PR TITLE
Feat/fault/duplicate message

### PR DESCRIPTION
**Pull request by symbiont-daniel-gustafsson**

### CODE REVIEW

File: src/executor-event-loop/executor.go [hunk 1]
```diff
@@ -129,7 +129,7 @@ func (el Executor) processEnvelope(env Envelope) (Message, ReactorsUpdateInfo) {
 	case "timer":
 		type TimerRequest struct {
 			Reactor string       `json:"to"`
-			At      time.Time    `json:"at"`
+			At      lib.TimePico `json:"at"`
```
Comment (symbiont-stevan-andjelkovic) [inline src/executor-event-loop/executor.go:132]: I think the change to the lib is missing?

File: src/runtime-prototype/src/Scheduler.hs [hunk 2]
```diff
@@ -125,19 +128,27 @@ resolveClientResponses _testId _runId _runInfo timeC _cr = do
...
+randomList NoRandom len = return $ replicate len 0
+randomList (WithMean mean) len = randomListOfExp len mean
+
+scheduleAgenda :: Time -> WithRandom -> [UnscheduledEvent] -> Free (ActorF s) Agenda
+scheduleAgenda t wr evs = do
+  ts <- map (addTime t . fromRational . toRational . realToFrac @Double @Nano)
+        <$> randomList wr (length evs)
+  let evs' = concat (zipWith toScheduled ts evs)
+  return $ Agenda.fromList (map (\e -> (at e, e)) evs')
+  where
```
Comment (symbiont-stevan-andjelkovic) [inline src/runtime-prototype/src/Scheduler.hs:143]: Empty where.

**Review by symbiont-stevan-andjelkovic — APPROVED** [review]

File: src/executor-event-loop/executor.go (hunk 1 above)
Comment (symbiont-daniel-gustafsson) [reply src/executor-event-loop/executor.go:132]: Fixed now, thanks

File: src/runtime-prototype/src/Scheduler.hs (hunk 2 above)
Comment (symbiont-daniel-gustafsson) [reply src/runtime-prototype/src/Scheduler.hs]: Fixed, thanks

**Pull request merged**